src/app/system/manage-data-tables/data-table.resolver.ts

```diff
@@ -0,0 +1,34 @@
+/** Angular Imports */
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+
+/** rxjs Imports */
+import { Observable } from 'rxjs';
+
+/** Custom Services */
+import { SystemService } from '../system.service';
+
+/**
+ * Data Table data resolver.
+ */
+@Injectable()
+export class DataTableResolver implements Resolve<Object> {
+
+  /**
+   * @param {SystemService} systemService System service.
+   */
+  constructor(private systemService: SystemService) {}
+
+  /**
+   * Returns the Data Table data.
+   * @returns {Observable<any>}
+   *
+   * // TODO: Delete the extra column to avoid multiple usages of `this.columnsData.shift()`.
```

Comment:
```suggestion
   * Returns the Data Table data.
   * TODO: Delete the extra column to avoid multiple usages of `this.columnsData.shift()`.
   * @returns {Observable<any>}
```